src/lib.rs

```diff
@@ -11,6 +11,898 @@
+                );
+
+                let (_, service_id) = split_u32_to_u16(source_filter.ue_id);
+                let instance_id = vsomeip::ANY_INSTANCE; // TODO: Set this to 1? To ANY_INSTANCE?
+                let (_, method_id) = split_u32_to_u16(source_filter.resource_id);
+
+                {
+                    let requested_services = REQUESTED_SERVICES.write().await;
+                    if !requested_services.contains(&(service_id, instance_id, method_id)) {
+                        get_pinned_application(application_wrapper).request_service(
+                            service_id,
+                            instance_id,
+                            ANY_MAJOR,
+                            ANY_MINOR,
+                        );
+                    }
+                }
+
+                trace!(
+                    "{}:{} - register_message_handler: service: {} instance: {} method: {}",
+                    UP_CLIENT_VSOMEIP_TAG,
+                    UP_CLIENT_VSOMEIP_FN_TAG_REGISTER_LISTENER_INTERNAL,
+                    service_id,
+                    instance_id,
+                    method_id
+                );
+
+                register_message_handler_fn_ptr_safe(
+                    application_wrapper,
+                    service_id,
+                    instance_id,
+                    method_id,
+                    msg_handler,
+                );
+
+                trace!(
+                    "{}:{} - Registered vsomeip message handler.",
+                    UP_CLIENT_VSOMEIP_TAG,
+                    UP_CLIENT_VSOMEIP_FN_TAG_REGISTER_LISTENER_INTERNAL,
+                );
+
+                Ok(())
+            }
+            RegistrationType::AllPointToPoint(_) => Err(UStatus::fail_with_code(
+                UCode::INTERNAL,
+                "Should be impossible to register point-to-point internally",
+            )),
+        }
+    }
+
+    async fn unregister_listener_internal(
+        source_filter: UUri,
+        sink_filter: Option<UUri>,
+        registration_type: RegistrationType,
+        application_wrapper: &UniquePtr<ApplicationWrapper>,
+        _runtime_wrapper: &UniquePtr<RuntimeWrapper>,
+    ) -> Result<(), UStatus> {
+        trace!(
+            "{}:{} - Attempting to unregister: source_filter: {:?} & sink_filter: {:?}",
+            UP_CLIENT_VSOMEIP_TAG,
+            UP_CLIENT_VSOMEIP_FN_TAG_REGISTER_LISTENER_INTERNAL,
+            source_filter,
+            sink_filter
+        );
+
+        match registration_type {
+            RegistrationType::Publish(_) => {
+                trace!(
+                    "{}:{} - Unregistering for Publish style messages.",
+                    UP_CLIENT_VSOMEIP_TAG,
+                    UP_CLIENT_VSOMEIP_FN_TAG_REGISTER_LISTENER_INTERNAL,
+                );
+                let (_, service_id) = split_u32_to_u16(source_filter.ue_id);
+                let instance_id = vsomeip::ANY_INSTANCE; // TODO: Set this to 1? To ANY_INSTANCE?
```

Comment:
What should this be set to? Wondering if you know @int0x27 / @stevenhartley

Reply:
same as above AFAIK